### PR TITLE
Support docker service names and links for PIHOLE_DNS_

### PR DESCRIPTION
**Pull request by haxwithaxe**
Added the ability to use docker service names or links in `PIHOLE_DNS_`.

## Description
In `start.sh` I added a check for valid domains to the block in the loop over `PIHOLE_DNS_`. The valid domains are resolved and the IP addresses are added to the configs in the same way as DNS servers specified as IPs. I also slightly rearranged the if block to avoid duplicating the warning about invalid servers.

## Motivation and Context 
I have a setup that uses an unbound container to provide DoT. I'm trying to integrate this composition as a stack in a swarm and it would greatly simplify the configuration if docker service names could be used. It would also make the behavior of this image more idiomatic. 

## How Has This Been Tested?
1. `ARCH=amd64 ./gh-actions-test.sh` 25 passed, 11 skipped
3. Using the docker-compose.yml  (below) with `PIHOLE_DNS_` set to "unbound;1.2.3.4" I started the composition.
4. I logged in to the pi-hole admin UI and verified the IP of the unbound container was set as the first custom server and 1.2.3.4 was the second. Then I removed the etc-* directories created in the working directory.
5. I repeated this with just IPv4 addresses.
6. I repeated this with just IPv6 addresses. The parts related to my change worked, but this produced an unrelated pihole-FTL error that exists in `pihole/pihole:latest` as well.
7. I repeated this with just the docker link.
Except for the IPv6 only test the specified IPs or IPs of specified service names were shown as the custom DNS servers and no errors were apparent in the logs.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project. (I tried to be consistent with the surrounding code)
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

```
version: "2"

services:
  pihole:
    container_name: pihole
    image: pihole:support-docker-links-for-PIHOLE_DNS_dev-amd64-buster
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      #- "67:67/udp"  # Required for DHCP
      - "80:80/tcp"
    environment:
      TZ: "America/New_York"
      PIHOLE_DNS_: "unbound;1.2.3.4"
    volumes:
      - "./etc-pihole:/etc/pihole"
      - "./etc-dnsmasq.d:/etc/dnsmasq.d"
    #cap_add:  # Required for DHCP
    #  - NET_ADMIN
    restart: unless-stopped

  unbound:
    image: haxwithaxe/unbound-dot:latest
```

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -127,9 +127,22 @@ if [ -n "${PIHOLE_DNS_}" ]; then
...
+          echo "Resolved ${i} from PIHOLE_DNS_ as: ${resolved_ip}"
+          if valid_ip "$resolved_ip" || valid_ip6 "$resolved_ip" ; then
+            change_setting "PIHOLE_DNS_$count" "$resolved_ip"
+            ((count=count+1))
+            ((valid_entries=valid_entries+1))
+            continue
+          fi
+        fi
+        # If the above tests fail then this is an invalid DNS server
+        echo "Invalid IP detected in PIHOLE_DNS_: ${i}"
```
Comment (PromoFaux) [inline start.sh:145]:
```suggestion
        echo "Invalid entry detected in PIHOLE_DNS_: ${i}"
```

Comment (PromoFaux) [reply start.sh:145]: While you are under the hood changing this, please could you change it inside the `if [ $valid_entries -eq 0 ]` block? thanks!

**Comment by PromoFaux** [comment]
Thanks for the submission (apologies it's taken a while to review)

Can you please address the DCO sign-off requirement to satisfy the bot? thanks! https://github.com/pi-hole/docker-pi-hole/pull/1014/checks?check_run_id=5485296936

File: start.sh (hunk 1 above)
Comment (haxwithaxe) [reply start.sh]: Fixed both in a new commit.

**Comment by haxwithaxe** [comment]
My last commit is broken :( Fixing now.

**Review by PromoFaux — APPROVED** [review]
Great stuff, thanks. Tested and working as described

**Comment by networkException** [comment]
Thanks a lot for implementing this! I can finally decomission my patched start.sh then 😅

**Pull request merged**